app/lib/conversation_details/group_details.dart

```diff
@@ -114,9 +106,63 @@ class GroupDetails extends StatelessWidget {
               },
               child: const Text("Add members"),
             ),
+            Divider(color: Theme.of(context).hintColor),
+            OutlinedButton(
+              style: dangerButtonStyle(context),
+              onPressed: () => _leave(context, conversation.id),
+              child: const Text("Leave"),
+            ),
+            OutlinedButton(
+              style: dangerButtonStyle(context),
+              onPressed: () => _delete(context, conversation.id),
+              child: const Text("Delete"),
+            ),
```

Comment:
We haven't fully specified this at the product level just yet, but just deleting a conversation if you haven't previously left it should not be possible.

I think the UX should be as follows:
 - In an active conversation, there's only the option to leave it.
 - If the leave operation is successful, the user should be prompted to ask if they want to delete the conversation as well, or keep it.
 - If they choose to delete, all state (including the MLS group state) should be deleted.
 - If they choose to keep the conversation, the leave button should o longer be shown. Instead, there should be a delete button.

Reply:
In addition, we should limit this to group conversations for now. Leaving a 1:1 conversation is UB right now; we need product specs first.